luigi/scheduler.py

```diff
@@ -200,12 +200,45 @@ def _get_default(x, default):
         return default
 
 
+class OrderedSet(object):
```

Comment:
I had written it myself but after thinking a bit more I've decided it would be better to use the recipe at http://code.activestate.com/recipes/576694/ which is mentioned at the bottom of the [collection docs](https://docs.python.org/2/library/collections.html).